ui/src/app/shared/services/workflows-service.ts

```diff
@@ -241,22 +241,43 @@ export const WorkflowsService = {
                     }
                     throw new Error('no artifact logs are available');
                 }
-
                 return from(requests.get(this.getArtifactLogsPath(workflow, nodeId, container, archived)));
             }),
             mergeMap(r => r.text.split('\n')),
+            filter(x => !!x), // check for empty content
             map(content => ({content, podName: workflow.status.nodes[nodeId].displayName}) as LogEntry),
             filter(x => !!x.content.match(grep))
         );
     },
 
+    getContainerLogsFromArtifacts(workflow: Workflow, container: string, grep: string, archived: boolean): Observable<LogEntry> {
+        // Iterate over all nodes and check if node is a pod. Then sort nodes by startedAt.
+        const podNodeIds: string[] = Object.keys(workflow.status.nodes)
+            .filter(nodeId => {
+                return workflow.status.nodes[nodeId].type === 'Pod';
+            })
+            .sort((a, b) => {
+                return new Date(workflow.status.nodes[a].startedAt).getTime() - new Date(workflow.status.nodes[b].startedAt).getTime();
+            });
+
+        // Iterate over pod node ids and get their logs
+        return from(podNodeIds).pipe(
+            concatMap(nodeId => {
+                return this.getContainerLogsFromArtifact(workflow, container, nodeId, container, grep, archived);
+            })
+        );
+    },
+
     getContainerLogs(workflow: Workflow, podName: string, nodeId: string, container: string, grep: string, archived: boolean): Observable<LogEntry> {
         const getLogsFromArtifact = () => this.getContainerLogsFromArtifact(workflow, nodeId, container, grep, archived);
+        const getLogsFromArtifacts = () => this.getContainerLogsFromArtifacts(workflow, container, grep, archived);
         const getLogsFromCluster = () => this.getContainerLogsFromCluster(workflow, podName, container, grep);
 
         // If our workflow is archived, don't even bother inspecting the cluster for logs since it's likely
         // that the Workflow and associated pods have been deleted
         if (archived) {
+            // If we don't have a node id, then we want the logs from all nodes
+            if (!nodeId) return getLogsFromArtifacts();
```

Comment:
A more optimal implementation may be to check that there is no `podName`.
There's a bug in the `workflow-details` page that a `podName` gets passed in when you click "Logs" at the top level (it's [the `nodeId`](https://github.com/argoproj/argo-workflows/blob/ec5b5d5f49d33172cd5dd018b4c5434551d93970/ui/src/app/workflows/components/workflow-details/workflow-details.tsx#L500), which used to be equivalent to the pod name prior to `POD_NAMES=v2` (#7279, #8748)), but when that's fixed, no `podName` would be equal to "All". Knowing that, we might be able to extend this feature to non-archived workflows that use archive logs.